File: backend/internal/dto/backtest.go
package dto

type BacktestRequest struct {
	Episode                string `json:"episode"`
	From                   string `json:"from"`
	To                     string `json:"to"`
	IncludeShap            bool   `json:"include_shap"`
	IncludeModuleBreakdown bool   `json:"include_module_breakdown"`
}

type BacktestMetric struct {
	Name  string  `json:"name"`
	Value float64 `json:"value"`
	Unit  string  `json:"unit"`
}

type BacktestEvent struct {
	Date        string  `json:"date"`
	Title       string  `json:"title"`
	Description string  `json:"description"`
	LSI         float64 `json:"lsi"`
	Status      string  `json:"status"`
}

type BacktestResponse struct {
	Episode              string               `json:"episode"`
	Range                DateRange            `json:"range"`
	LSIHistory           []LSIHistoryPoint    `json:"lsi_history"`
	Metrics              []BacktestMetric     `json:"metrics"`
	Events               []BacktestEvent      `json:"events"`
	AverageContributions []ModuleContribution `json:"average_contributions"`
	Conclusion           string               `json:"conclusion"`
}

File: frontend/src/shared/types/backtest.ts
import type { ModuleContribution } from "@/shared/types/dashboard";
import type { LsiHistoryPoint } from "@/shared/types/lsi";

export interface BacktestMetric {
  name: string;
  value: number | string;
  unit?: string;
}

export interface BacktestEvent {
  date?: string;
  title?: string;
  description?: string;
  severity?: string;
}

export interface BacktestRange {
  from: string;
  to: string;
}

export interface BacktestResponse {
  episode: string;
  range: BacktestRange;
  lsi_history: LsiHistoryPoint[];
  metrics: BacktestMetric[];
  events: BacktestEvent[];
  average_contributions: ModuleContribution[];
  conclusion: string;
}

export interface BacktestParams {
  episode?: string;
  from?: string;
  to?: string;
  include_shap?: boolean;
  include_module_breakdown?: boolean;
}
